src/app/core/parsers/zod-entity-parser.service.ts

```diff
@@ -0,0 +1,75 @@
+import { EntityParserService } from '../abstractions/entity-parser.service'
+import { AdLibPiece } from '../models/ad-lib-piece'
+import { BasicRundown } from '../models/basic-rundown'
+import { Part } from '../models/part'
+import { Piece } from '../models/piece'
+import { Rundown } from '../models/rundown'
+import { Segment } from '../models/segment'
+import * as zod from 'zod'
+
+
+export class ZodEntityParser implements EntityParserService {
```

Comment:
`EntityParserService` is an abstract class, so you should `extend` it right? Not `implements` it?

Reply:
The only reason it is an abstract class instead of an interface is to have a runtime representation of the interface.

Reply:
Doesn't matter why it's an abstract class. It's still an abstract class ;)